src/debug_input.cpp

```diff
@@ -79,95 +66,29 @@ DebugInput::DebugInput(Window &window, SceneManager &scene_manager, Physics &p)
         } else if (d.action == GLFW_RELEASE) {
             keys[d.key] = false;
         }
+
+        if (debug_input_on)
+            debug_movement_keys();
+
     });
 
     events::window_cursor_event.connect([&](events::WindowCursorData d) {
         int width, height;
         std::tie(width, height) = d.window->get_size();
         glm::vec3 current_cursor_pos(d.x_pos, d.y_pos, 0);
 
-        Scene *scene = scene_manager.get_current_scene();
-        SceneCamera *camera = scene_manager.get_camera();
-
-        // First movement detected.
-        if (!mouse_moved) {
-            mouse_moved = true;
-            last_cursor_pos = current_cursor_pos;
-            return;
-        }
-
-        glm::vec3 cursor_delta = current_cursor_pos - last_cursor_pos;
-        if (lmb_down && keys[GLFW_KEY_LEFT_CONTROL]) {
-            int dir = cursor_delta.x > 0 ? 1 : -1;
-            float rot_angle = dir * glm::length(cursor_delta) * 0.001f;
-            scene->light_pos = glm::vec3(
-                    glm::rotate(glm::mat4(1.0f), rot_angle, glm::vec3(0.f, 0.f, 1.f)) * glm::vec4(scene->light_pos, 1.0f));
-        } else if (!keys[GLFW_KEY_LEFT_CONTROL]) {
-            // Look around.
-            GLfloat xoffset = cursor_delta.x;
-            GLfloat yoffset = cursor_delta.y;
-            GLfloat sensitivity = 0.5;
-            xoffset *= sensitivity;
-            yoffset *= sensitivity;
-
-            if (current_cursor_pos.x > width - EDGE_PAN_THRESH)
-                xoffset = EDGE_PAN_SPEED;
-            else if (current_cursor_pos.x < EDGE_PAN_THRESH)
-                xoffset = -EDGE_PAN_SPEED;
-
-            camera->yaw += xoffset;
-            camera->pitch += yoffset;
-
-            if (camera->pitch > 89.0f)
-                camera->pitch = 89.0f;
-            if (camera->pitch < -89.0f)
-                camera->pitch = -89.0f;
-
-            glm::vec3 front;
-            front.x = cos(glm::radians(camera->yaw)) * cos(glm::radians(camera->pitch));
-            front.y = -sin(glm::radians(camera->pitch));
-            front.z = sin(glm::radians(camera->yaw)) * cos(glm::radians(camera->pitch));
-            camera->cam_front = glm::normalize(front);
-            camera->recalculate();
-        }
-
-        physics.raycast_mouse(d.x_pos, d.y_pos, width, height);
+        if (debug_input_on)
+            debug_movement_mouse(d);
 
         last_cursor_pos = current_cursor_pos;
     });
 
     events::window_mouse_button_event.connect([&](events::WindowMouseButtonData d) {
```

Comment:
Should probably just get rid of the entire `connect` if we don't need it. Ditto for the `window_scroll_event` one.